presto-native-execution/presto_cpp/main/PeriodicHeartbeatManager.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "presto_cpp/main/PeriodicHeartbeatManager.h"
+#include <velox/common/memory/Memory.h>
+
+namespace facebook::presto {
+PeriodicHeartbeatManager::PeriodicHeartbeatManager(
```

Comment:
There is a lot of similarity with announcer. I was thinking having a base class for retry delay, http (client creation and https) and only override status (json) creation. What do you think?